git/git.go

```diff
@@ -9,111 +9,113 @@ import (
 )
 
 var execCommand = exec.Command
-var clone = Clone
-var setConfig = SetConfig
-var mergePR = MergePR
-var fetchPR = FetchPR
-var merge = Merge
-var reset = Reset
 
-// command executes the git command
-func command(arguments ...string) error {
-	cmd := execCommand("git", arguments...)
-
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	err := cmd.Start()
-	if err != nil {
-		return fmt.Errorf("starting git command: %v", err)
-	}
+type repo struct {
+	ScmURL string
+	Path   string
+	Branch string
+}
 
-	err = cmd.Wait()
-	if err != nil {
-		return fmt.Errorf("running git command: %v", err)
-	}
-	return nil
+// Repo is a git repo
+type Repo interface {
+	Checkout(sha string) error
+	MergePR(prNumber string) error
 }
 
-// Clone clones a git repo into a destination directory
-func Clone(scmURL, destination string) error {
-	log.Printf("Cloning %v", scmURL)
+// New returns a new repo object
+func New(scmURL, path string) (Repo, error) {
 	parts := strings.Split(scmURL, "#")
 	if len(parts) < 2 {
-		return fmt.Errorf("expected #branchname in SCM URL: %v", scmURL)
+		return nil, fmt.Errorf("expected #branchname in SCM URL: %v", scmURL)
 	}
-
-	repo := parts[0]
-	branch := parts[1]
-	return command("clone", "--quiet", "--progress", "--branch", branch, repo, destination)
-}
-
-// SetConfig sets up git configuration
-func SetConfig(setting, name string) error {
-	return command("config", setting, name)
-}
-
-// FetchPR fetches a pull request into a specified branch
-func FetchPR(prNumber string, branch string) error {
-	return command("fetch", "origin", "pull/"+prNumber+"/head:"+branch)
-}
-
-// Merge merges changes on the specified branch
-func Merge(branch string) error {
-	return command("merge", "--no-edit", branch)
+	repo := repo{
+		ScmURL: parts[0],
```

Comment:
should we add const values for the 0 and 1 magic numbers?

Something like "SCM_URL_INDEX" and "BRANCH_INDEX"